scripts/control_bar.js

```diff
@@ -2,21 +2,121 @@
 /* exported makeImageWidget */
 
 // Construct the image sizing controls.
-var makeImageControl = function(imageElement) {
+var makeImageControl = function(canvas, reSize) {
     let imageKey;
-    let percent;
-    const percentInput = $("<input>", {type: 'number', min: '1', max: '999', value: percent, title: texts.zoomPercent});
+    let scale = 1;
+    let imageWidth, imageHeight;
+    const image = document.createElement("img");
+    let ctx = canvas.getContext("2d");
+
+    const percentInput = $("<input>", {type: 'number', min: '1', max: '999', value: 100, title: texts.zoomPercent});
+
+    let sine = 0, cosine = 1;
+
+    let scrollbarWidth = false;
+    var getScrollbarWidth = function() {
+        if (scrollbarWidth === false) {
+            let testDiv = document.createElement('div');
+            testDiv.innerHTML = '<div style="width:50px;height:5px;position:absolute; overflow:scroll;"><div style="width:100px;height:5px;"></div></div>';
```

Comment:
Will this cause a flash of a scroll bar on slow browsers? I wonder if there is a solution where we don't need to know scrollbar width.

Reply:
I haven't managed to find such a solution.